### DISCUSSION TITLE
Accessibility enhancements with Qt6 (TT5.11)

### DISCUSSION THREAD

**Discussion by CoBC** [BearWare/TeamTalk5#1499] [General]
Hi to everyone,
this discussion is here to inform you about furure improvements in TT5.11 (currently in beta), thanks to Qt6.
## On Windows
1. Empty trees or trees with only one item are now announced without requiring press on arrow key.
2. "Expanded" or "Collapsed" state is now natively announce (work arround has been removed).
3. "Checked" or "Unchecked" on some trees like TTS events settings is now natively announce (work arround has been removed).

## MacOS
1. Trees and list are now natively accessible using arrow keys (option to speak list has been removed).

Users who are still using classic client are one more time encouraged to switch to Qt client.
You can discussed here about accessibility on TT5.11 (Qt6), I already reported some issues to Qt and/or NVAccess teams.
Thanks for your contribution.

**Comment by alessio0292** [comment]
is a problem for me because i use seven.

**Comment by va3ets200** [comment]
I'm running the latest beta version that's out there, version 5.11.0.5095 on Mac OS on a 2020 intel MacBook air.  Using VoiceOver of course.  When trying to delete a server from my server list, (These are servers that I've entered manually and save, not official, unofficial, or public servers,)  hitting the remove entry button, which I think should do it, does nothing at all, the server I tried to remove,  ends up still being there.  Additionally, I cannot accomplish the task of deleting user accounts on a server that I have, on which I'm administrator.  On the Windows client from what I've been told, there is a delete button there in the user accounts window.  I'm selecting the user account that I want to delete, just by arrowing to it in the list of user accounts.  If there is a delete button on screen, it is not at all visible to VoiceOver.  IF these two fairly significant accessibility issues could be fixed in the Mac app, prior to the full release of version 5.11, that would be great.  As it is frustrating, having these old servers that are no longer up in my server list, as well as not being able to delete user accounts.

**Comment by CoBC** [comment]
@bear101 Context menu is not accessible for now on MacOS using 
VoiceOver. Do you think you can re-add the Delete button on user 
accounts and server list dialog on MacOS only?

**Reply by bear101** [reply]
Hm, yes, but we'll have the same problem in "User Accounts" dialog. Here it's also context menu for deleting.

**Reply by bear101** [reply]
Ah, saw you already mentioned that. I wonder if it's easy to create a keyboard shortcut, so, say F12 will show context menu if control is focused.

**Comment by CoBC** [comment]
Apparently the problem is not to display context menu, but to interact 
with it, VO don't see context menu items currently, Qt devs semms to be 
working on a fix but nothing available yet.

**Comment by va3ets200** [comment]
Yeah now that would be nice. To be focussed on a server and or a user account, and have it be deleted just at the press of a shortcut keySent from my iPhoneOn Sep 27, 2022, at 4:55 PM, Bjørn Damstedt Rasmussen ***@***.***> wrote:﻿
Ah, saw you already mentioned that. I wonder if it's easy to create a keyboard shortcut, so, say F12 will show context menu if control is focused.

**Comment by va3ets200** [comment]
For now, is there a pList file somewhere that I  can delete, to totally clear out my server list?  I'd really like to see these two issues of deleting servers which you've manually added, and also deleting user accounts resolved.  It's totally doable from what I'm told on the Windows client, but I don't have any windows machines, and am not getting one any time soon.

**Comment by CoBC** [comment]
On V5.11, you'll be able to delete user or server just using Delete key.

**Reply by va3ets200** [reply]
Nice.  Just got the latest beta update, as I have mine set to check for beta updates, and this is working absolutely splendidly.  Thanks a lot for this.

**Comment by va3ets200** [comment]
Agreed, that's how it should be.
